### ISSUE TITLE
Basic Battle Agent Code

### ISSUE THREAD

**Issue by kev-lamb** [kev-lamb/mgba_LuaToPython#5]

**Comment by kev-lamb** [comment]
Thoughts: since battles have a continuous state space, something like basic Q-learning won't work here. Doing a neural network on its own is also a bit weird because we want to incorporate short term and long term rewards. I think the best approach is to perform function approximation using a neural network, then using some sort of loss w.r.t the bellman equation to update the network after the conclusion of a battle. 

Should be able to do this with pytorch or tensorflow. Worst case, I think i might have some custom code for this from a past project

**Comment by kev-lamb** [comment]
fully implemented code for training a Deep Q network for battling. When battles end, a random battle from the training set is selected
